pkg/services/pluginsintegration/clientmiddleware/metrics_middleware.go

```diff
@@ -26,21 +28,26 @@ type pluginMetrics struct {
 type MetricsMiddleware struct {
 	pluginMetrics
 	pluginRegistry registry.Service
+	features       featuremgmt.FeatureToggles
 	next           plugins.Client
 }
 
-func newMetricsMiddleware(promRegisterer prometheus.Registerer, pluginRegistry registry.Service) *MetricsMiddleware {
+func newMetricsMiddleware(promRegisterer prometheus.Registerer, pluginRegistry registry.Service, features featuremgmt.FeatureToggles) *MetricsMiddleware {
+	var additionalLabels []string
+	if features.IsEnabled(featuremgmt.FlagPluginsInstrumentationStatusSource) {
+		additionalLabels = []string{"status_source"}
+	}
 	pluginRequestCounter := prometheus.NewCounterVec(prometheus.CounterOpts{
 		Namespace: "grafana",
 		Name:      "plugin_request_total",
 		Help:      "The total amount of plugin requests",
-	}, []string{"plugin_id", "endpoint", "status", "target"})
+	}, append([]string{"plugin_id", "endpoint", "status", "target"}, additionalLabels...))
 	pluginRequestDuration := prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Namespace: "grafana",
 		Name:      "plugin_request_duration_milliseconds",
 		Help:      "Plugin request duration",
 		Buckets:   []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 25, 50, 100},
-	}, []string{"plugin_id", "endpoint", "target"})
+	}, append([]string{"plugin_id", "endpoint", "target"}, additionalLabels...))
```

Comment:
i think `plugin_request_duration_seconds` metric below is the important one to include the label for. If we add it for these metrics as well guess doesn't hurt

Reply:
Ah yes, I agree. Thanks for reminding, I think it's worth involving enterprise datasources to double check which metrics they would need at first :+1: